library/src/layout/par.rs

```diff
@@ -457,22 +457,33 @@ impl<'a> Line<'a> {
         self.items().skip(start).take(end - start)
     }
 
-    /// How many justifiable glyphs the line contains.
+    /// How many glyphs are in the text where we can insert additional
+    /// space when encountering underfull lines.
     fn justifiables(&self) -> usize {
         let mut count = 0;
+        let mut last_is_justifiable = false;
         for shaped in self.items().filter_map(Item::text) {
-            count += shaped.justifiables();
+            let (justifiable_count, last) = shaped.justifiables();
+            count += justifiable_count;
+            last_is_justifiable = last;
```

Comment:
why if the last item is not a text item? does that work correctly with `last_is_justifiable`?